### ISSUE TITLE
Crash on "Manage Recordings"

### ISSUE THREAD

**Issue by AndyJRobinson** [bennettpeter/android-MythTV-Leanfront#31]
The app crashes when selecting "Manage Recordings". There must be something in the data coming from mythbackend, but I can't figure out exactly what. The system log shows

04-30 09:52:49.747 11213 11213 D AndroidRuntime: Shutting down VM
04-30 09:52:49.752 11213 11213 E AndroidRuntime: FATAL EXCEPTION: main
04-30 09:52:49.752 11213 11213 E AndroidRuntime: Process: org.mythtv.leanfront, PID: 11213
04-30 09:52:49.752 11213 11213 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {org.mythtv.leanfront/org.mythtv.leanfront.ui.ManageRecordingsActivity}: java.lang.NumberFormatException: For input string: ""
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3822)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3854)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:193)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6718)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: Caused by: java.lang.NumberFormatException: For input string: ""
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at java.lang.Integer.parseInt(Integer.java:627)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at java.lang.Integer.parseInt(Integer.java:650)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at z3.f.k2(Unknown Source:169)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at z3.f.o2(Unknown Source:23)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at z3.f.L0(Unknown Source:3)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.Fragment.k1(Unknown Source:16)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.T0(Unknown Source:791)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.Q0(Unknown Source:93)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.R0(Unknown Source:42)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.j0(Unknown Source:4)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.h0(Unknown Source:6)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.Fragment.k1(Unknown Source:41)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.T0(Unknown Source:791)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.Q0(Unknown Source:93)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.R0(Unknown Source:42)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.j0(Unknown Source:4)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.i.h0(Unknown Source:6)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.e.p(Unknown Source:4)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.c.v(Unknown Source:9)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at androidx.fragment.app.c.onPostResume(Unknown Source:3)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.app.Activity.performResume(Activity.java:7325)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3814)
04-30 09:52:49.752 11213 11213 E AndroidRuntime: 	... 11 more
04-30 09:52:49.762  2847  3863 W ActivityManager:   Force finishing activity org.mythtv.leanfront/.ui.ManageRecordingsActivity
04-30 09:52:49.763  2847  3863 I UsageStatsService: paused component:org.mythtv.leanfront

**Comment by AndyJRobinson** [comment]
Found the issue. At some point during a mythfilldatabase run, there was a channum with only alphabetic characters added to the channel table. That causes the parseInt in GridFragment.java to throw an exception:

```
            if (chanNumStr != null) {
                String[] chanNumSplit = chanNumStr.split("[^0123456789]");
                if (chanNumSplit.length > 0 && chanNumSplit[0] != null)
                    chanNum = Integer.parseInt(chanNumSplit[0]);
            }
```

**Comment by bennettpeter** [comment]
Thank you for debugging this for me!

**Comment by bennettpeter** [comment]
Fixed in build 20220509-leanfront-release-v0-354-gfba8737
